aiida/manage/tests/main.py

```diff
@@ -348,6 +348,7 @@ def create_profile(self):
         if created:
             user.store()
         profile.default_user_email = user.email
+        config.store()
```

Comment:
just curious how this came up in this PR?
another process reading the config?

Reply:
I wanted to actually deal with this in a separate PR to properly test for the fix, but I went down a rabbit hole of having to refactor the test infrastructure and fixtures, so I postponed that. Might have a look later still

The problem is that if you create temporary test profile from scratch, this method is called. As you can see, the default user is created and then set on the config. However, this is just done in memory and the config wasn't persisted to disk. If the daemon was then started, it would read a config without default user defined, and so attempting to create any node (which would try to load the default user to set as the node's owner) would except.

Reply:
That makes sense. Might be worth adding a comment above, like 
```suggestion
        # needed e.g. for tests that involve the daemon
        config.store()
```